src/attended-event/attended-event.service.spec.ts

```diff
@@ -0,0 +1,187 @@
+import { Test, TestingModule } from "@nestjs/testing";
+import { AttendedEventService } from "./attended-event.service";
+import { TestDatabaseModule } from "../test-database/test-database.module";
+import { MongooseModule } from "@nestjs/mongoose";
+import { UserService } from "../user/user.service";
+import { EventService } from "../event/event.service";
+import { User } from "../user/interfaces/user.interface";
+import { Event } from "../event/interfaces/event.interface";
+import { AttendedEventSchema } from "./schemas/attended-event.schema";
+
+class MockUserService {
+  async create() {
+    /* Should be overriden using spyOn */
+  }
+  async findAll() {
+    /* Should be overriden using spyOn */
+  }
+  async findByAuthId() {
+    /* Should be overriden using spyOn */
+  }
+  async update() {
+    /* Should be overriden using spyOn */
+  }
+}
+
+class MockEventService {
+  async create() {
+    /* Should be overriden using spyOn */
+  }
+  async findAll() {
+    /* Should be overriden using spyOn */
+  }
+  async findById() {
+    /* Should be overriden using spyOn */
+  }
+  async update() {
+    /* Should be overriden using spyOn */
+  }
+}
+
+describe("AttendedEventService", () => {
+  let service: AttendedEventService;
+  let userService: UserService;
+  let eventService: EventService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      imports: [
+        TestDatabaseModule,
+        MongooseModule.forFeature([
+          { name: "AttendedEvent", schema: AttendedEventSchema }
+        ])
+      ],
+      providers: [
+        AttendedEventService,
+        { provide: UserService, useValue: new MockUserService() },
+        { provide: EventService, useValue: new MockEventService() }
+      ]
+    }).compile();
+
+    service = module.get<AttendedEventService>(AttendedEventService);
+    userService = module.get<UserService>(UserService);
+    eventService = module.get<EventService>(EventService);
+  });
+
+  it("should be defined", () => {
+    expect(service).toBeDefined();
+    expect(userService).toBeDefined();
+    expect(eventService).toBeDefined();
+  });
+
+  it("should record an AttendedEvent given a valid userAuthId and eventId", async () => {
+    jest.spyOn(userService, "findByAuthId").mockImplementation(
+      async (): Promise<User> =>
+        Promise.resolve({
```

Comment:
As the mock implementation function is `async`, wouldn't it be better to just return the object itself and let the function wrap it in a promise?

Something like:
```
async (): Promise<User> => ({
          authId: "random",
          email: "example@example.com",
          notificationEmail: "example@example.com"
        } as User)
```